### DISCUSSION TITLE
kBeforeCreateReference callback for USD

### DISCUSSION THREAD

**Discussion by chadmv** [Autodesk/maya-usd#2481] [General]
Maya has a kBeforeCreateReferenceCheck callback which allows me to sync any files from version control (Perforce) before the file is referenced which greatly speeds up user workflow. Is there any similar callback functionality with USD so I can sync all other referenced USD files before they are opened in Maya?

**Comment by charlesfleche** [comment]
Not directly, but I think a custom [ArResolver](https://graphics.pixar.com/usd/release/api/class_ar_resolver.html) could be used for such a purpose.
